### DISCUSSION TITLE
Custom rendering backend?

### DISCUSSION THREAD

**Discussion by Absolucy** [emilk/egui#719] [Q&A]
How could I add  a custom rendering backend for egui? For example, using libogc to render on a Wii.

**Comment by coderedart** [comment]
https://docs.rs/egui/0.14.2/egui/struct.CtxRef.html#method.end_frame is where you start the rendering part. after that you call https://docs.rs/egui/0.14.2/egui/struct.CtxRef.html#method.tessellate to get a batches of triangles that you need to render. you can look at how each of the different windowing/rendering libraries list at https://github.com/emilk/egui#3rd-party .  egui itself only needs input like the framebuffer size or events etc.. with the https://docs.rs/egui/0.14.2/egui/struct.RawInput.html and outputs the stuff that needs to be drawn with end_frame. everything else should be managed by you with your backend. epi mostly uses the official backend by glium, look at the App trait to see what you need to implement to work with epi. https://docs.rs/epi/0.14.0/epi/backend/index.html
